core/services/vrf/vrfcommon/inflight_cache.go

```diff
@@ -0,0 +1,82 @@
+package vrfcommon
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type InflightCache interface {
+	Add(lg types.Log)
+	Contains(lg types.Log) bool
+	Size() int
+}
+
+var _ InflightCache = (*inflightCache)(nil)
+
+const cachePruneInterval = 1000
+
+type inflightCache struct {
+	// cache stores the logs whose fulfillments are currently in flight or already fulfilled.
+	cache map[logKey]struct{}
+
+	// lookback defines how long state should be kept for. Logs included in blocks older than
+	// lookback may or may not be redelivered.
+	lookback int
+
+	// lastPruneHeight is the blockheight at which logs were last pruned.
+	lastPruneHeight uint64
+
+	// mu synchronizes access to the delivered map.
+	mu sync.Mutex
```

Comment:
It looks like a `RWMutex` would be useful in this case
```suggestion
	mu sync.RWMutex
```